src/main/kotlin/com/tally/kiosk/common/base/BaseTimeEntity.kt

```
@@ -0,0 +1,22 @@
+import org.springframework.data.annotation.CreatedDate
+import org.springframework.data.annotation.LastModifiedDate
+import org.springframework.data.jpa.domain.support.AuditingEntityListener
+import java.time.LocalDateTime
+
+@MappedSuperclass
+@EntityListeners(AuditingEntityListener::class)
+abstract class BaseTimeEntity protected constructor() {
+    @CreatedDate
+    @Column(updatable = false)
+    lateinit var createdAt: LocalDateTime
+        protected set
```

detekt (1.23.7)

[warning] Usages of lateinit should be avoided.

(detekt.potential-bugs.LateinitUsage)